core/trino-main/src/main/java/io/trino/operator/aggregation/AggregationMask.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.aggregation;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.block.RunLengthEncodedBlock;
+
+import javax.annotation.Nullable;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+
+public final class AggregationMask
+{
+    private int positionCount;
+    private boolean selectAll;
+    private int[] selectedPositions = new int[0];
+    private int selectedPositionCount;
+
+    public AggregationMask()
+    {
+        reset(0);
+    }
+
+    public void reset(int positionCount)
+    {
+        this.positionCount = positionCount;
+        this.selectedPositionCount = 0;
+        this.selectAll = true;
+    }
+
+    public int getPositionCount()
+    {
+        return positionCount;
+    }
+
+    public boolean isSelectAll()
+    {
+        return selectAll;
+    }
+
+    public boolean isSelectNone()
+    {
+        return positionCount == 0 || (!selectAll && selectedPositionCount == 0);
+    }
+
+    public int getSelectedPositionCount()
+    {
+        checkState(!selectAll, "selectedPositionCount not available when in selectAll mode");
+        return selectedPositionCount;
+    }
+
+    public int[] getSelectedPositions()
+    {
+        checkState(!selectAll, "getSelectedPositions not available when in selectAll mode");
+        return selectedPositions;
+    }
+
+    public void unselectNullPositions(Block block)
+    {
+        unselectPositions(block, false);
+    }
+
+    public void applyMaskBlock(@Nullable Block maskBlock)
+    {
+        if (maskBlock != null) {
+            unselectPositions(maskBlock, true);
+        }
+    }
+
+    private void unselectPositions(Block block, boolean testValue)
```

Comment:
The name of the `testValue` argument confused me at first. I interpreted it as "the value to use for testing", but it appears to be a flag for indicating "whether values should be tested".

Maybe rename to `shouldTestValue`, `evaluateValues`, `testValues`, or something along those lines for clarity.